monads/test/Spec.hs

```diff
@@ -1,2 +1,49 @@
+import Test.Tasty
+import Test.Tasty.HUnit
+import Text.Printf (printf)
+import Control.Monad (unless)
+import qualified Data.Map as Map
+import HW.StackMachine
+import HW.Eval
+
+-- Numeric operations: Push numbers and test basic arithmetic
+numCases :: [(StackProgram String, Either (Error String) (MachineState String))]
+numCases =
+  [ ([PushNum 5], Right (MachineState [5] Map.empty))
+  , ([PushNum 0], Right (MachineState [0] Map.empty))
+  , ([PushNum (-10)], Right (MachineState [-10] Map.empty))
```

Comment:
In general, it's pointless to test the same scenario with different numbers, when the evaluator behaves the same on each of them.